### ISSUE TITLE
Sometimes error forbidden

### ISSUE THREAD

**Issue by davidkroell** [abrander/garmin-connect#2]
Hi, @abrander 

I've build some CLI with your library, and I sometimes get the error `forbidden` using the exact same credentials.
Do you know why this is the case? Does a workaround exists?

Some thoughts in this issue: https://github.com/davidkroell/bodycomposition/issues/6

Regards,
David

**Comment by abrander** [comment]
Thanks for reporting the issue.

First of all, are you using the newest version? Newer than 5d9e621984 - that commit added support for some changes made by Garmin that would otherwise result in spurious sign-in failures.

The library can output all HTTP communication and some logging by initializing like this:

```go
connect.NewClient(
    connect.Credentials(email, password),
    connect.DumpWriter(os.Stderr),
    connect.LogWriter(log.New(os.Stderr, "", log.LstdFlags)),
)
```

That output with credentials removed would be very helpful in debugging these issues.

**Comment by davidkroell** [comment]
I tried this, but `connect.LogWriter()` does not seem to work.

Nevertheless, I figured out how to write a logfile. I do not really detect something strange, expecting

```json
{"message":"HTTP 403 Forbidden","error":"WebApplicationException"}
```

Sometimes it also returns an `HTTP 429 Too Many Requests` statuscode.

I upload the log as file, because this is kinda big.
[dumpfile.log](https://github.com/abrander/garmin-connect/files/5314415/dumpfile.log)

**Comment by abrander** [comment]
Garmin will answer 429 after too many failed requests (replies in the 4xx range).

I think I can replicate. I'll have a look - in the meantime, please make sure that you're using the newest version of this package. `go get -u github.com/abrander/garmin-connect`.

**Comment by abrander** [comment]
You log provided everything to debug this. Thanks. Should be fixed in 1123b5b69c - can you confirm?

**Comment by davidkroell** [comment]
Works like a charm, awesome!
